pkg/osbuild/gzip_stage_test.go

```diff
@@ -0,0 +1,47 @@
+package osbuild
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewGzipStageOptions(t *testing.T) {
+	filename := "image.raw.gz"
+
+	expectedOptions := &GzipStageOptions{
+		Filename: filename,
+	}
+
+	actualOptions := NewGzipStageOptions(filename)
+	assert.Equal(t, expectedOptions, actualOptions)
+}
+
+func TestNewGzipStage(t *testing.T) {
+	inputFilename := "image.raw"
+	filename := "image.raw.gz"
+	pipeline := "os"
+
+	expectedStage := &Stage{
+		Type:    "org.osbuild.gzip",
+		Options: NewGzipStageOptions(filename),
```

Comment:
Using Options: &GzipStageOptions{Filename: filename}, here feels slightly better (its a bit of tradeoff of course). The reason is that if we (accidentally) change NewGzipStageOptions to e.g. switch to a default compression level of say "5" with this test setup no test would fail but with the explicit GzipStageOptions{} we would get an error.

Reply:
Mmm, this was purely for consistency with the others; I think we'll probably end up dropping the compression level.

Reply:
No need to change anything here, I can make a note to myself to do a followup if needs changes in more places. But I still wanted to mention it because it highlights a general principle: testing the result of  NewGzipStageOptions() with the result of NewGzipStageOptions() in the test will test less than using the &GzipStageOptions{} directly here (unless of course we already check elsewhere that NewGzipStageOptions() has the expected result). But please ignore for now, this is also very theoretical because the NewGzipOptions() thing is so small.